test/integration/script-loader/test/index.test.js

```diff
@@ -163,21 +163,31 @@ const runTests = (isDev = false) => {
     })
   }
 
-  if (!isDev) {
-    it('onload fires correctly', async () => {
-      let browser
-      try {
-        browser = await webdriver(appPort, '/page4')
-        await waitFor(3000)
+  it('onload fires correctly', async () => {
+    let browser
+    try {
+      browser = await webdriver(appPort, '/page4')
+      await waitFor(3000)
 
-        const text = await browser.elementById('text').text()
+      const text = await browser.elementById('onload-div-1').text()
+      expect(text).toBe('aaabbbccc')
 
-        expect(text).toBe('aaabbbccc')
-      } finally {
-        if (browser) await browser.close()
-      }
-    })
+      // Navigate to different page and back
+      await browser.waitForElementByCss('[href="/page9"]')
+      await browser.click('[href="/page9"]')
+      await browser.waitForElementByCss('[href="/page4"]')
+      await browser.click('[href="/page4"]')
```

Comment:
Add a navigation test. This is to test that `onLoad` should not execute after navigation away and back.